[Bug] FE error during upgrade if new publish is enabled

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
2.5 enables new publish by default, which may cause FE error during upgrade.

### Expected behavior (Required)

### Real behavior (Required)

```
2022-12-07 14:22:23,035 WARN (replayer|84) [GlobalStateMgr$5.runOneCycle():1666] got interrupt exception or inconsistent exception when replay journal 10071375, will exit,
com.starrocks.journal.JournalInconsistentException: failed to load journal type 100
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:926) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1806) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1663) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.run(GlobalStateMgr.java:1728) [starrocks-fe.jar:?]
Caused by: java.lang.IllegalStateException: only can remove node with no dependency
        at com.google.common.base.Preconditions.checkState(Preconditions.java:510) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.transaction.TransactionGraph.remove(TransactionGraph.java:107) ~[starrocks-fe.jar:?]
        at com.starrocks.transaction.DatabaseTransactionMgr.unprotectUpsertTransactionState(DatabaseTransactionMgr.java:1172) ~[starrocks-fe.jar:?]
        at com.starrocks.transaction.DatabaseTransactionMgr.replayUpsertTransactionState(DatabaseTransactionMgr.java:1589) ~[starrocks-fe.jar:?]
        at com.starrocks.transaction.GlobalTransactionMgr.replayUpsertTransactionState(GlobalTransactionMgr.java:615) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:518) ~[starrocks-fe.jar:?]
        ... 4 more
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

--- a/fe/fe-core/src/main/java/com/starrocks/transaction/TransactionGraph.java
+++ b/fe/fe-core/src/main/java/com/starrocks/transaction/TransactionGraph.java
@@ -77,6 +77,11 @@ public class TransactionGraph {
         public int hashCode() {
             return Objects.hash(txnId);
         }
+
+        @Override
+        public String toString() {
+            return Long.toString(txnId);
+        }
     }
 
     private Map<Long, Node> nodes = new HashMap<>();
@@ -117,7 +122,12 @@ public class TransactionGraph {
         if (node == null) {
             return;
         }
-        Preconditions.checkState(node.ins == null || node.ins.isEmpty(), "only can remove node with no dependency");
+        if (node.ins != null && !node.ins.isEmpty()) {
+            LOG.warn("remove txn " + txnId + " with dependency: " + node.ins + " this may happen during FE upgrading");
+            for (Node dep : node.ins) {
+                dep.outs.remove(node);
+            }
+        }
         nodes.remove(txnId);
         nodesWithoutIns.remove(node);
         for (long tableId : node.writeTableIds) {

--- a/fe/fe-core/src/test/java/com/starrocks/transaction/TransactionGraphTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/transaction/TransactionGraphTest.java
@@ -54,6 +54,24 @@ public class TransactionGraphTest {
     }
 
     @Test
+    public void testRemoveNodeWithDependency() {
+        TransactionGraph graph = new TransactionGraph();
+        graph.add(1, Lists.newArrayList(1L));
+        graph.add(2, Lists.newArrayList(2L));
+        graph.add(3, Lists.newArrayList(1L));
+        graph.add(4, Lists.newArrayList(2L));
+        graph.add(5, Lists.newArrayList(1L));
+        graph.add(6, Lists.newArrayList(2L));
+        assertEquals(graph.size(), 6);
+        graph.remove(3);
+        graph.remove(4);
+        assertEquals(graph.size(), 4);
+        expectNextBatch(graph, Lists.newArrayList(1L, 2L, 5L, 6L));
+        assertEquals(graph.size(), 0);
+        assertEquals(graph.getTxnsWithoutDependency().size(), 0);
+    }
+
+    @Test
     public void testMultiTableTxn() {
         TransactionGraph graph = new TransactionGraph();
         graph.add(1, Lists.newArrayList(1L));

